modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/utils/client_extensions/loadEditorClientExtensions.ts

```diff
@@ -24,7 +24,7 @@ export default function loadEditorClientExtensions({
 				})
 			),
 			onLoad: (bindingContexts) => {
-				let transformedConfig = config;
+				let transformedConfig = JSON.parse(JSON.stringify(config));
```

Comment:
The intended behavior we were going for is that the CXs modification to editor config is a local change. Original editor configuration, a controlled variable, should remain in it's original form. But because we are making an assignment here, changes leak to initial config object. @dsanz your intuition was right on this one! The shallow copy is also not good enough, as it would leak changes (JS :hugs: ) . We need to do a deep copy here. cc @thektan

The issue still remains that deep in parent framework config object changes on actions that should not really change it, such as focus on the field. This causes re-initialization of the whole editor, when it really doesn't need to. Improving this could be difficult, and is ultimately in the domain of product team. Because of this, I didn't revert the solution we did in https://github.com/liferay-frontend/liferay-portal/pull/4430. This is the glitchy behavior if we do the revert on top of this PR:

https://github.com/user-attachments/assets/1e370bc4-3e75-4167-b4dc-95e019360954

:warning:  Although solution after this PR behaves nice and covers existing cases, there is a hidden problem: **if original configuration legitimately changes, through some future action in UI, CXs will not be re-applied**. I believe this is something that would be impossible for us to fix. If such a scenario appears, we will need to revert solution in 4430, and (1) live with glitchy behavior or (2) refactor parent framework to update config object only in some cases.

Reply:
I see.... assignment makes 2 variables pointing to the same thing. One of the variables mutates the value.